enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,340 @@
+spike across all clusters might give us early warning to a cloud provider’s data
+center outage, or a large network or similar event that is impacting many of our
+users. Having a total number of remediation events can demonstrate the value
+our platform is bringing to our users. It will show the operational burden
+we’ve decreased on teams that use OpenShift.
+
+For example, since many clusters operate at a relatively small scale, it’s
+possible that losing an Availability Zone in AWS US-East-1 region would be
+small enough event on their cluster to trigger the Machine Health Check. Letting
+users know we’re seeing this across many clusters would assist in their own troubleshooting
+efforts and save them time.
+
+**Remediation is Disabled or Short Circuited**
+
+Epics: Proactive Support
+
+This metric will give us insight into when a particular cluster has too many
+unhealthy nodes for the Machine Health Check to do its job. A user may be expecting
+that the health check is running as normal, cleaning up broken hosts, when in
+reality it has been silently disabled. Any sustained disabling of the Machine
+Health Check should potentially trigger a proactive support engagement. Similar
+to the Total Remediation Events, experiencing a high volume of these events
+across several users or platforms provides opportunities to create alerts for
+users of potential outage events.
+
+**Number of Nodes Covered by the Machine Health Check**
+
+Epics: Engineering Insights, Proactive Support
+
+This metric will give us insight into user experiences operating the Machine Health
+Check. For example, users might think they are utilizing the Machine Health
+Check correctly only to find out that it has been misconfigured and not
+performing its role. If users have a large cluster with the health check
+enabled, but not covering any nodes, this might be an opportunity to reach out
+and see if we can help with their configuration. This can also help us identify
+areas where increased education will have value.
+
+**Failed Node Cleanups**
+
+Epics: Proactive Support
+
+This metric will help us identify conditions where the Cluster Autoscaler is
+silently cleaning up failied node creations. We should trigger an alert if the
+autoscaler cleans up more than X hosts in Y time period. This alert would be
+useful for users that might not realize they’re having these infrastructure problems.
+This is another case where signals across multiple clusters might indicate
+that a cloud or network area is having problems.
+
+For example, our internal continuous integration infrastructure team had a
+situation where two MachineSets had been configured for autoscaling.
+One MachineSet would scale up without issue, then the other would not. While
+this was happening, the Machines would fail, alarms would fire, and the autoscaler
+would dispose of the nodes that failed to come online. These failed nodes were
+being silently cleaned up by the autoscaler.
+
+**Total Scale Events**
```

Comment:
I am missing in all of these the actual metric that will be used here. Are these new metrics that you will introduce, or existing ones?  


Not sure you seen but we have telemetry docs for each team, how to measure number of series, the amount of series to be sent etc. https://docs.google.com/document/d/1a6n5iBGM2QaIQRg9Lw4-Npj6QY9--Hpx3XYut-BrUSY/edit

Reply:
> I am missing in all of these the actual metric that will be used here. Are these new metrics that you will introduce, or existing ones?

i've tried to break the implementation section into a couple parts, this first part talks about how we will use the exported metrics and what calculations we would like to determine from them. in the following section "Alerts, Metrics, and Telemetry" i start to enumerate the specific metrics we will need to export. i can add some details about specific metric names there, would that work?

Reply:
It's up to you, we can also do it in the PRs, happy to review those, or here. There are certain best practices around metric naming. 
>  i can add some details about specific metric names there, would that work?

Its more about are these new metrics or are you using existing metrics to send. So this would be helpful to use actual metric names yes. It's more about to answer questions like:
- could these metrics be done upstream
- maybe we have these metrics already, so we don't duplicate the efforts
- best practices around naming these metrics if they are new 
- are these going to be high unbound cardinality metrics

hence examples of actual metrics and all their possible labels help with both in cluster prometheus ingestion as well as keeping limit of series for those we sending to telemetry.

Reply:
> Its more about are these new metrics or are you using existing metrics to send. So this would be helpful to use actual metric names yes. It's more about to answer questions like:
> 
>     * could these metrics be done upstream
> 
>     * maybe we have these metrics already, so we don't duplicate the efforts
> 
>     * best practices around naming these metrics if they are new
> 
>     * are these going to be high unbound cardinality metrics

thanks, this is good advice.

i think i talk about _some_ of these details below, but i will go back and make sure that it is clear we are answering these questions.